extensions/reviewed/Gamepads.json

```diff
@@ -1390,12 +1387,171 @@
+    },
+    {
+      "description": "Generate an advanced vibration on the specified controller. Incompatible with Firefox.",
+      "fullName": "Advanced gamepad vibration",
+      "functionType": "Action",
+      "name": "A_Advanced_Vibration_Controller",
+      "sentence": "Make gamepad _PARAM1_ vibrate for _PARAM2_ seconds with the vibration magnitude of _PARAM3_ and _PARAM4_",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "/** @type {Gamepad[]} */",
+            "//Vibration work only on game in browser.",
+            "const gamepads = navigator.getGamepads ? navigator.getGamepads() : (navigator.webkitGetGamepads ? navigator.webkitGetGamepads() : []);",
+            "",
+            "//Get function parameters",
+            "const playerId = eventsFunctionContext.getArgument(\"Player_ID\") - 1;",
+            "const duration = eventsFunctionContext.getArgument(\"Duration\") || 1;",
+            "const strongRumbleMagnitude = eventsFunctionContext.getArgument(\"StrongMagnitude\");",
+            "const weakRumbleMagnitude = eventsFunctionContext.getArgument(\"WeakMagnitude\");",
+            "",
+            "if (playerId < 0 || playerId > 4) {",
+            "  console.error('Parameter gamepad identifier in action: \"Advanced gamepad vibration\", is not valid number, must be between 0 and 4.');",
+            "  return;",
+            "}",
+            "if (weakRumbleMagnitude < 0 || weakRumbleMagnitude > 1) {",
+            "  console.error('Parameter weakRumble identifier in action: \"Advanced gamepad vibration\", is not valid number, must be between 0 and 1.');",
+            "  return;",
+            "}",
+            "if (strongRumbleMagnitude < 0 || strongRumbleMagnitude > 1) {",
+            "  console.error('Parameter strongRumble identifier in action: \"Advanced gamepad vibration\", is not valid number, must be between 0 and 1.');",
+            "  return;",
+            "}",
+            "",
+            "const gamepad = gamepads[playerId];",
+            "",
+            "//we need keep this condition because when use have not yet plug the controller we can't get the controller in the gamepad variable.",
+            "if (gamepad == null) return;",
+            "",
+            "if (gamepad && gamepad.vibrationActuator) {",
+            "  gamepad.vibrationActuator.playEffect(\"dual-rumble\", {",
+            "    startDelay: 0,",
+            "    duration: duration * 1000,",
+            "    weakMagnitude: weakRumbleMagnitude,",
+            "    strongMagnitude: strongRumbleMagnitude",
+            "  });",
+            "}",
+            "",
+            "gdjs._extensionController.rumble.duration = duration;",
+            "gdjs._extensionController.rumble.elapsedTime = 0;",
+            "gdjs._extensionController.rumble.weakMagnitude = weakRumbleMagnitude;",
+            "gdjs._extensionController.rumble.strongMagnitude = strongRumbleMagnitude;"
```

Comment:
These attributes are affected in an action that focus on 1 gamepad, but they override the same attribute for any gamepad. It should use an array with `playerId` as index.